### PR TITLE
matrix_summ

### PR DESCRIPTION
**Pull request by Ypostolov**

### CODE REVIEW

File: YELLOW_BELT/week_1/matrix_summ.cpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+#include <iostream>
+#include <stdexcept>
+#include <vector>
+
+class Matrix {
+ public:
+  int num_rows_;
+  int num_columns_;
+  int** arr;
+  std::vector<std::vector<int> > elements_;
```
Comment (ZhdanovP) [inline YELLOW_BELT/week_1/matrix_summ.cpp]: @Ypostolov We discussed to do it via the array, not vector

**Review by ZhdanovP — CHANGES_REQUESTED** [review]

File: YELLOW_BELT/week_1/matrix_summ.cpp (hunk 1 above)
Comment (Ypostolov) [reply YELLOW_BELT/week_1/matrix_summ.cpp]: Fixed in c049e8ef94431348aec0ca0e79069a828d8fc1ba.
Please check.

File: YELLOW_BELT/week_1/matrix_summ.cpp [hunk 2]
```diff
@@ -1,107 +1,41 @@
...
-
-  int GetNumRows() const {
-    return num_rows_;
-  }
-
-  int GetNumColumns() const {
-    return num_columns_;
+  ~Matrix() {
+    for (int i = 0; i < num_rows_; ++i) {
+      delete arr[i];
```
Comment (ZhdanovP) [inline YELLOW_BELT/week_1/matrix_summ.cpp]: @Ypostolov No right deleting.
Note! You are deleting two dimensional **array**

File: YELLOW_BELT/week_1/matrix_summ.cpp [hunk 3]
```diff
@@ -1,107 +1,41 @@
...
-  for (int row = 0; row < matrix.GetNumRows(); ++row) {
-    for (int column = 0; column < matrix.GetNumColumns(); ++column) {
-      if (column > 0) {
-        out << " ";
-      }
-      out << matrix.At(row, column);
-    }
-    out << std::endl;
+  void show(int row_number_, int column_number_) {
+    std::cout << arr[row_number_][column_number_] << " ";
```
Comment (ZhdanovP) [inline YELLOW_BELT/week_1/matrix_summ.cpp]: @Ypostolov Let's do show not a method but a function. And implement method int get(int i, int j) to get corespondent element from array

File: YELLOW_BELT/week_1/matrix_summ.cpp (hunk 2 above)
Comment (Ypostolov) [reply YELLOW_BELT/week_1/matrix_summ.cpp]: Fixed in 9204b5c6dab6fc2f9ebd652d8302c06680e23c15.
Please check!

File: YELLOW_BELT/week_1/matrix_summ.cpp (hunk 3 above)
Comment (Ypostolov) [reply YELLOW_BELT/week_1/matrix_summ.cpp]: Fixed in 9204b5c6dab6fc2f9ebd652d8302c06680e23c15
Please check.

**Comment by ZhdanovP** [comment]
@Ypostolov Well, let's do not constant size matrix.
We need possibility to pass matrix size via constructor.
Example of using : Matrix m(2, 5); which will created matrix 2x5


@ZhdanovP  
Implemented in f139dcdb2583205491c63ffaeaeff32f2211a709.
Please take a look!

File: YELLOW_BELT/week_1/matrix_summ.cpp [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+      }
+    }
+  }
+
+  return true;
+}
+
+Matrix operator+(const Matrix& one, const Matrix& two) {
+  if (one.GetNumRows() != two.GetNumRows()) {
+    throw std::invalid_argument("Mismatched number of rows");
```
Comment (ZhdanovP) [inline YELLOW_BELT/week_1/matrix_summ.cpp]: @Ypostolov Let's do operators not like a function but as a methods

Comment (Ypostolov) [reply YELLOW_BELT/week_1/matrix_summ.cpp]: Implemented in 48c7880cc746ba42d26ce65dee9acb43da80b64e.
Please take a look.

**Comment by ZhdanovP** [comment]
@Ypostolov Great. Now let is do Matrix as a template class
@ZhdanovP  Implemented in b37b6b0a6a5b075229b855ecb86f6fd38442106e. Please take a look.

**Comment by ZhdanovP** [comment]
@Ypostolov It is not complete realization of template class. You still is storing int instead of template type
@ZhdanovP  implemented  in 06fb1fb3cc07603ebca1b9d643904f992a08742f. Please take a look.

**Pull request merged**